### DISCUSSION TITLE
Difference between BigInt sizes in nodejs and databases

### DISCUSSION THREAD

**Discussion by saymynameplsvip** [prisma/prisma#22589] [Q&A]
I am forced to store 64 bit numbers in a database (MySQL). In MySQL, such numbers are considered as BigInt, and in NodeJS as Int. When I receive such a number from the database, prisma interprets it as a BigInt, and the third-party libraries to which I need to pass this number interpret it as number. This requires introducing some tricks into the code (converting bigint to string and then to number).

I propose to match data types by size in bits.
```[tasklist]
### Tasks
```

**Comment by SevInf** [comment]
JS has no `Int` type, aside from `BigInt`. `number` is double-precision float and can store integers without losing any information only up to 53 bits. If we are to convert 64-bit DB integers to JS numbers, we will silently lose precision for bigger numbers.

**Reply by BitRayDev** [reply]
This is true, but if the business logic of the application operates with the `number` type and assumes values in the `number` range, Prisma should not force the user to use `BigInt` and rewrite the business logic for `BigInt`. This is a real problem, it should be possible to map the `Int` type to `@db.BigInt` in schema.

**Reply by BitRayDev** [reply]
This topic has already been discussed here: #7570, many people face this problem and it needs to be solved.

**Reply by SevInf** [reply]
There is no way for us to know that your application does not actually use full 64bit range. All we know that DB field can go up to 64 bit and JS `number` can't store such values, so we can't map it to `number` for everybody.

If that's the case for you, you may implement [result extension](https://www.prisma.io/docs/orm/prisma-client/client-extensions/result) for your model that would convert `BigInt` to `number` automatically, bearing all responsibility for potential data loss coming from this conversion.

**Reply by BitRayDev** [reply]
Using Result Extension is not convenient and still requires quite a lot of code overhead, this functionality should be available "out of the box".

It is not necessary to map BigInt to number for everybody. It should just be possible to specify `@db.BigInt` for `Int` type in the schema. As a protection against accidental specification, it is possible to throw warning, or to name this type `@db.DangerousBigInt`

ORM should simplify, not complicate. In this case it only complicates.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
